sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala

```diff
@@ -126,4 +140,59 @@ object FileSourceStrategy extends Strategy with Logging {
 
     case _ => Nil
   }
+
+  private def generateStructFieldsContainsNesting(
```

Comment:
You may make this method `private[sql]` so that you don't need to rely on the `PrivateMethod` ScalaTest trick to test it.